[padding-line-between-statements] Documented `cjs-import` and `cjs-export` statement types are not accepted

[The documentation](https://eslint.style/rules/default/padding-line-between-statements#rule-details) of the `padding-line-between-statements` rule states that `cjs-import` and `cjs-export` are valid statement types.

However, linting the following JS file with the following command results in an error.

```js
const path = require('node:path');
module.exports = {};
```

```console
gebsh@gebsh % eslint --no-eslintrc --env es2023 --plugin '@stylistic' --rule '@stylistic/padding-line-between-statements: [error, { blankLine: always, prev: cjs-import, next: cjs-export }]' ./test.cjs

Oops! Something went wrong! :(

ESLint: 8.53.0

Error: CLIOptions:
	Configuration for rule "@stylistic/padding-line-between-statements" is invalid:
	Value "cjs-import" should be equal to one of the allowed values.
	Value "cjs-import" should be array.
	Value "cjs-import" should match some schema in anyOf.
```

I believe this is caused by the fact that `@typescript-eslint` renames these two options to `require` and `exports`: https://github.com/typescript-eslint/typescript-eslint/issues/4091. Renaming them in the configuration does work:

```console
gebsh@gebsh % eslint --no-eslintrc --env es2023 --plugin '@stylistic' --rule '@stylistic/padding-line-between-statements: [error, { blankLine: always, prev: require, next: exports }]' ./test.cjs

<project-root>/test.cjs
  2:1  error  Expected blank line before this statement  @stylistic/padding-line-between-statements

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

--- a/packages/eslint-plugin-ts/rules/padding-line-between-statements/padding-line-between-statements.test.ts
+++ b/packages/eslint-plugin-ts/rules/padding-line-between-statements/padding-line-between-statements.test.ts
@@ -5124,5 +5124,13 @@ declare namespace Types {
       options: [{ blankLine: 'always', prev: '*', next: 'block-like' }],
       errors: [{ messageId: 'expectedBlankLine' }],
     },
+
+    // https://github.com/eslint-stylistic/eslint-stylistic/issues/53
+    {
+      code: `const path = require('node:path');\nmodule.exports = {};`,
+      output: `const path = require('node:path');\n\nmodule.exports = {};`,
+      options: [{ blankLine: 'always', prev: 'cjs-import', next: 'cjs-export' }],
+      errors: [{ messageId: 'expectedBlankLine' }],
+    },
   ],
 })

--- a/packages/eslint-plugin-ts/rules/padding-line-between-statements/padding-line-between-statements.ts
+++ b/packages/eslint-plugin-ts/rules/padding-line-between-statements/padding-line-between-statements.ts
@@ -13,6 +13,9 @@ import {
   createRule,
 } from '../../utils'
 
+const CJS_EXPORT = /^(?:module\s*\.\s*)?exports(?:\s*\.|\s*\[|$)/u
+const CJS_IMPORT = /^require\(/u
+
 /**
  * This rule is a replica of padding-line-between-statements.
  *
@@ -569,6 +572,18 @@ const StatementTypes: Record<string, NodeTestObject> = {
     'while',
   ),
   'with': newKeywordTester(AST_NODE_TYPES.WithStatement, 'with'),
+
+  'cjs-export': {
+    test: (node, sourceCode) => node.type === 'ExpressionStatement'
+    && node.expression.type === 'AssignmentExpression'
+    && CJS_EXPORT.test(sourceCode.getText(node.expression.left)),
+  },
+  'cjs-import': {
+    test: (node, sourceCode) => node.type === 'VariableDeclaration'
+    && node.declarations.length > 0
+    && Boolean(node.declarations[0].init)
+    && CJS_IMPORT.test(sourceCode.getText(node.declarations[0].init!)),
+  },
 
   // Additional Typescript constructs
   'interface': newKeywordTester(

--- a/packages/eslint-plugin-ts/rules/padding-line-between-statements/types.d.ts
+++ b/packages/eslint-plugin-ts/rules/padding-line-between-statements/types.d.ts
@@ -41,6 +41,8 @@ export type StatementType =
     | 'var'
     | 'while'
     | 'with'
+    | 'cjs-export'
+    | 'cjs-import'
     | 'interface'
     | 'type'
     )
@@ -84,6 +86,8 @@ export type StatementType =
         | 'var'
         | 'while'
         | 'with'
+        | 'cjs-export'
+        | 'cjs-import'
         | 'interface'
         | 'type'
       ),
@@ -126,6 +130,8 @@ export type StatementType =
         | 'var'
         | 'while'
         | 'with'
+        | 'cjs-export'
+        | 'cjs-import'
         | 'interface'
         | 'type'
       )[],

